src/transformers/generation/tf_utils.py

```diff
@@ -2927,15 +2930,15 @@ def beam_search(
         >>> # lets run beam search using 3 beams
         >>> num_beams = 3
         >>> # define decoder start token ids
-        >>> input_ids = tf.ones((num_beams, 1), dtype=tf.int64)
+        >>> input_ids = tf.ones((1, num_beams, 1), dtype=tf.int32)
         >>> input_ids = input_ids * model.config.decoder_start_token_id
 
         >>> # add encoder_outputs to model keyword arguments
-        >>> model_kwargs = {
-        ...     "encoder_outputs": model.get_encoder()(
-        ...         tf.repeat(encoder_input_ids, num_beams, axis=0), return_dict=True
-        ...     )
-        ... }
+        >>> encoder_outputs = model.get_encoder()(encoder_input_ids, return_dict=True)
+        >>> encoder_outputs.last_hidden_state = tf.repeat(
+        ...     tf.expand_dims(encoder_outputs.last_hidden_state, axis=0), num_beams, axis=1
+        ... )
+        >>> model_kwargs = {"encoder_outputs": encoder_outputs}
```

Comment:
Not looking the details here yet (I believe in you @gante) , but I don't know why this differs from what we have in `generation_utils.py`, where they look like
```python
        >>> # add encoder_outputs to model keyword arguments
        >>> model_kwargs = {
        ...     "encoder_outputs": model.get_encoder()(
        ...         encoder_input_ids.repeat_interleave(num_beams, dim=0), return_dict=True
        ...     )
        ... }
```
If the change here is necessary, does this mean we have somehow different APIs between our PyTorch/TF generation?

Reply:
Yes, it is a great point!

The two frameworks expect different input shapes at the moment *for beam search only*. This is because TF XLA beam search was borrowed from FLAX, which implements beam search with the beam dimension as a separate dimension. This is opposed to the old TF beam search and PT beam search, which pack the beam dimension with the batch dimension.

Contrastive search had the same problem of having a new beam dimension, but I figured out how to make it XLA compatible while keeping the same structure as in PT (i.e. packing the beam dimension along the batch dimension). In the near future, I want to push a similar change to TF beam search, which would make us get rid of this mismatch :) And more importantly, it should make TF XLA beam search faster, as we would get rid of many reshape operations!